docs/deployment/on_kubernetes.md

```diff
@@ -4,4 +4,112 @@
 
 </div>
 
-# Running Kyuubi on Kubernetes
+# Deploy Kyuubi engines on Kubernetes
+
+## Requirements
+
+When you want to run Kyuubi's Spark SQL engines on Kubernetes, you'd better have cognition upon the following things.
+
+* Read about [Running Spark On Kubernetes](http://spark.apache.org/docs/latest/running-on-kubernetes.html)
+* An active Kubernetes cluster
+* [Kubectl](https://kubernetes.io/docs/reference/kubectl/overview/)
+* KubeConfig of the target cluster
+
+## Configurations
+
+### Master
+
+Spark on Kubernetes config master by using a special format.
+
+`k8s://https://<k8s-apiserver-host>:<k8s-apiserver-port>`
+
+You can use cmd `kubectl cluster-info` to get api-server host and port.
+
+### Spark Image
+
+Kubernetes requires users to supply images that can be deployed into containers within pods.
+
+Spark also ships within a `./bin/docker-image-tool.sh` script that can be used to build and publish the Docker images to use
+with the K8s backend.
```

Comment:
```suggestion
Kubernetes requires users to supply images that can be deployed into containers within pods.

```suggestion
Spark ships a `./bin/docker-image-tool.sh` script to build and publish the Docker images for running Spark applications on Kubernetes.

When deploying Kyuubi engines against a Kubernetes cluster, we need to set up the docker images in the Docker registry first.
```